### PR TITLE
Correctly update App status on some deployment failures

### PR DESCRIPTION
**Pull request by uvegla**
Towards: https://github.com/giantswarm/giantswarm/issues/22017

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: service/controller/chart/resource/status/create.go [hunk 1]
```diff
@@ -35,6 +35,8 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	releaseName := key.ReleaseName(cr)
 	r.logger.Debugf(ctx, "getting status for release %#q", releaseName)
 
+	r.logger.Debugf(ctx, "the controller context for %#q is: %#q", releaseName, cc)
```
Comment (kubasobon) [inline service/controller/chart/resource/status/create.go:38]: Is it useful when debugging the Chart CR?

File: service/controller/chart/resource/status/create.go [hunk 2]
```diff
@@ -73,27 +75,32 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
...
 		}
 	}
 
 	desiredStatus := v1alpha1.ChartStatus{
-		AppVersion: releaseContent.AppVersion,
-		Reason:     reason,
-		Release: v1alpha1.ChartStatusRelease{
-			Revision: to.IntP(releaseContent.Revision),
-			Status:   status,
+		AppVersion: releaseContent.AppVersion, // The desired should depend on the version set in the Chart, not what is deployed by Helm
```
Comment (kubasobon) [inline service/controller/chart/resource/status/create.go:97]: There are so many questions here in the comments. Do you want to merge it like that, or would you rather wait for them to get answered?

Comment (uvegla) [reply service/controller/chart/resource/status/create.go:97]: Right, no it is a leftover if making sense of it all.

File: service/controller/chart/resource/status/create.go (hunk 1 above)
Comment (uvegla) [reply service/controller/chart/resource/status/create.go]: Nope, it will end up in the status reason now, cleaned it up.

**Review by MarcelMue — APPROVED** [review]

**Pull request merged**